h2o-algos/src/main/java/hex/gam/MatrixFrameUtils/GamUtils.java

```diff
@@ -146,6 +146,13 @@ public static void copy2DArray(double[][] src_array, double[][] dest_array) {
               src_array[colIdx].length);
     }
   }
+  
+  // copy a square array
+  public static double[][] copy2DArray(double[][] src_array) {
+    double[][] dest_array = new double[src_array.length][src_array.length];
```

Comment:
Shouldn't this use the MemoryManager? Also it seems to me that `copy2DArray(double[][], double[][])` supports any 2d array so I'd prefer not limiting the functionality without changing the name. So either I would change the memory allocation like this:
```java
double[][] dest_array = new double[src_array.length][];
for(int i = 0; i < src_array.length; ++i)
    dest_array[i] = MemoryManager.malloc8d(src_array[i].length);
```

or just rename it to `copySquareArray` or something like that. It still might be worth considering changing the memory allocation to use the MemoryManager. (I still don't know the code well enough to know if it copies just small matrices or even the big ones but if you think it's ok to have this memory allocation that's good enough for me).

Reply:
I added the change but kept the name because I want to copy rectangular arrays too.